### ISSUE TITLE
Minimum Emacs version?

### ISSUE THREAD

**Issue by KingCode** [plexus/chemacs2#42]
Thanks for making Chemacs2! 

My current version of Emacs is 27.2 and as it does provide for the use of `early-init.el` I assumed that Chemacs2 would work with it. 

But as hard as I tried, I wasn't able to make the `--with-profile` option work, or get emacs to pickup the right profile without specifying 'default' (either in .emacs-profile.el or within .emacs-profile):
```
$ cat ~/.emacs-profiles.el
(("legacy" . ((user-emacs-directory . "~/.config/emacs/.legacy")))
 ("current" . ((user-emacs-directory . "~/.config/emacs/.legacy")))
 ("rational" . ((user-emacs-directory . "~/.rational-emacs"))))

$ cat ~/.emacs-profile
cat: .emacs-profile: No such file or directory

$ emacs --with-profile legacy
```
The above results in  `error: No profile ‘default’ in ~/.emacs-profiles.el` within the emacs start-up screen 

However when either specifying 'default' within .emacs-profiles.el or e.g. 'legacy' within .emacs-profiles, emacs always picks the default/legacy screen as per above, no matter what `--with-profile` specifies:
```
$ echo 'legacy' > ~/.emacs-profile
 $ emacs --with-profile rational
```
(results in emacs `legacy` profile instead of `rational`)

So the only way to make Emacs 27.2 choose the right version is by hard-coding the chosen profile within `.emacs-profile`
before each start-up, which is not definitely not as fluid as intended by Chemacs2 :)

However, after compiling and trying out emacs 29.0.50, Chemacs2 works great and as intended.

I assume I should install Emacs 28 - can anyone confirm a Minimum version before I do?
Thanks!

**Comment by plexus** [comment]
I feel like 27 should work, I'm pretty sure original chemacs predates 27, so it clearly can work. If it doesn't I would say that's a bug and we should look into what is causing it.

**Comment by KingCode** [comment]
Sorry for the delay...after looking into it some more, and learning how to 'debug' the startup process, I sprinkled some message statements within `chemacs.el`: it turns out that the `command-line-args` value passed to `chemacs-handle-command-line` (invoked by the definition of `chemacs--with-profile-value` is the path to Emacs itself:
```
command-line-args is a variable defined in ‘C source code’.

Its value is
("/Applications/Emacs.app/Contents/MacOS/Emacs-x86_64-10_14")

Args passed by shell to Emacs, as a list of strings.
Many arguments are deleted from the list as they are processed.

  Probably introduced at or before Emacs version 17.
```
The `(message ...)` statements I sprinkled confirm the above. 

So it sounds like my command-line arguments intended for chemacs2 are overwritten by the shell somehow, and rejected by chemacs in favor of my environment variable default value $CHEMACS_PROFILE.

This happens with Emacs 28 release. When using Emacs 29 chemacs works like a charm.
 
I will digg further but any suggestion is appreciated - thanks!
